gdal/frmts/nitf/nitffile.c

```diff
@@ -2479,7 +2489,7 @@ static char** NITFGenericMetadataReadTREInternal(char **papszMD,
                             CPLSPrintf("%s%s", pszMDPrefix, "NPLN"));
                 int NPLN = atoi(NITFFindValFromEnd(papszMD, *pnMDSize, pszMDItemName, "-1"));
                 CPLFree(pszMDItemName);
-                if (NPLN < 0)
+                if (NPLN < 1)
```

Comment:
STDI-0002-1 indicates the valid range for this values is 2-999.  Changing this check guarantees that NPLN-1 is non-negative when we cast to unsigned long.